### PR TITLE
[RFR] upgrade_path not user writable

### PR DESCRIPTION
**Pull request by StefanRijnhart**
Fixing the security issue when apriori.py is imported from the upgrade-path, and the the user is allowed to set the upgrade-path in the interface.

### CODE REVIEW

**Comment by StefanRijnhart** [comment]
About the apriori.py in this module, I'm removing it because I don't understand why it's there (and it confused me)

File: upgrade_analysis/models/ir_module_module.py [hunk 1]
```diff
@@ -24,6 +24,9 @@ def _compute_is_oca_module(self):
     def _compute_is_odoo_module(self):
         for module in self:
             module_path = get_module_path(module.name)
+            if not module_path:
```
Comment (legalsylvain) [inline upgrade_analysis/models/ir_module_module.py:27]: thanks ! I didn't handled that case !

**Review by legalsylvain — APPROVED** [review]
> About the apriori.py in this module, I'm removing it because I don't understand why it's there (and it confused me)

Just for information : 
Sorry for not having explained that point. In the idea of the refactoring, there was the possibility to use this module for minor / custom upgrade. (ref [initial comment](https://github.com/OCA/OpenUpgrade/issues/2440#issuecomment-721680931), _"if a developper decides in V14, to refactor fully a module, he could install the old version V14.X on a instance, and the new version V14.Y on another instance, and run the analysis to see all the difference"_), so not in an "Openupgrade context". In that _very theoritical case_ user doesn't have to pull openupgrade repository, (only server-tools repo). for that reason I imagined the double ``apriori.py`` files in both modules, with one (``upgrade_analysis``) importing the second (``openupgrade_scripts``), with the possibility in the first file, to write "custom refactor".

But : 
- that's a very theoritical case
- it's maybe not the best design.

So, thanks for that PR.

Can I merge it ?

**Comment by StefanRijnhart** [comment]
Ah thanks for clarifying. I think for that theoretical case, maybe a list of mappings in the upgrade.analysis form might be helpful rather than having this second file in the code. Please go a head with the merge!

**Pull request merged**